geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/ExportConfigCommandDUnitTest.java

```diff
@@ -0,0 +1,181 @@
+    this.managerConfigFile = this.temporaryFolder.newFile("Manager-cache.xml");
+    this.managerPropsFile = this.temporaryFolder.newFile("Manager-gf.properties");
+    this.vm1ConfigFile = this.temporaryFolder.newFile("VM1-cache.xml");
+    this.vm1PropsFile = this.temporaryFolder.newFile("VM1-gf.properties");
+    this.vm2ConfigFile = this.temporaryFolder.newFile("VM2-cache.xml");
+    this.vm2PropsFile = this.temporaryFolder.newFile("VM2-gf.properties");
+    this.shellConfigFile = this.temporaryFolder.newFile("Shell-cache.xml");
+    this.shellPropsFile = this.temporaryFolder.newFile("Shell-gf.properties");
+    this.subDir = this.temporaryFolder.newFolder(getName());
+    this.subManagerConfigFile = new File(this.subDir, this.managerConfigFile.getName());
+  }
+
+  @Category(FlakyTest.class) // GEODE-1449
+  @Test
+  public void testExportConfig() throws Exception {
+    Properties localProps = new Properties();
+    localProps.setProperty(NAME, "Manager");
+    localProps.setProperty(GROUPS, "Group1");
+    setUpJmxManagerOnVm0ThenConnect(localProps);
+
+    // Create a cache in another VM (VM1)
+    Host.getHost(0).getVM(1).invoke(new SerializableRunnable() {
+      public void run() {
+        Properties localProps = new Properties();
+        localProps.setProperty(NAME, "VM1");
+        localProps.setProperty(GROUPS, "Group2");
+        getSystem(localProps);
+        getCache();
+      }
+    });
+
+    // Create a cache in a 3rd VM (VM2)
+    Host.getHost(0).getVM(2).invoke(new SerializableRunnable() {
+      public void run() {
+        Properties localProps = new Properties();
+        localProps.setProperty(NAME, "VM2");
+        localProps.setProperty(GROUPS, "Group2");
+        getSystem(localProps);
+        getCache();
+      }
+    });
+
+    // Create a cache in the local VM
+    localProps = new Properties();
+    localProps.setProperty(NAME, "Shell");
+    getSystem(localProps);
+    Cache cache = getCache();
+
+    // Test export config for all members
+    deleteTestFiles();
+    CommandResult cmdResult =
+        executeCommand("export config --dir=" + this.temporaryFolder.getRoot().getAbsolutePath());
+    assertEquals(Result.Status.OK, cmdResult.getStatus());
+
+    assertTrue(this.managerConfigFile + " should exist", this.managerConfigFile.exists());
+    assertTrue(this.managerPropsFile + " should exist", this.managerPropsFile.exists());
+    assertTrue(this.vm1ConfigFile + " should exist", this.vm1ConfigFile.exists());
+    assertTrue(this.vm1PropsFile + " should exist", this.vm1PropsFile.exists());
+    assertTrue(this.vm2ConfigFile + " should exist", this.vm2ConfigFile.exists());
+    assertTrue(this.vm2PropsFile + " should exist", this.vm2PropsFile.exists());
+    assertTrue(this.shellConfigFile + " should exist", this.shellConfigFile.exists());
+    assertTrue(this.shellPropsFile + " should exist", this.shellPropsFile.exists());
+
+    // Test exporting member
```

Comment:
I didn't know this at the time, but there is actually a separate JIRA ticket for refactoring all of the tests associated with all of these gfsh commands (see [GEODE-1359](https://issues.apache.org/jira/browse/GEODE-1359)). I'm going to roll back the changes to these tests and leave them for future work.